daemons/controld/controld_execd.c

```diff
@@ -731,17 +742,21 @@ build_operation_update(xmlNode * parent, const lrmd_rsc_info_t *rsc,
                                          controld_metadata_from_agent
                                          |controld_metadata_from_cache);
     if (metadata == NULL) {
-        return TRUE;
+        return;
     }
 
     crm_trace("Including additional digests for %s:%s:%s",
               rsc->standard, rsc->provider, rsc->type);
     append_restart_list(op, metadata, xml_op, caller_version);
     append_secure_list(op, metadata, xml_op, caller_version);
 
-    return TRUE;
+    return;
 }
 
+#define controld_add_resource_history_xml(parent, rsc, op, node_name)   \
+    controld_add_resource_history_xml_as(__func__, (parent), (rsc),     \
+                                         (op), (node_name))
```

Comment:
Depends on what you mean ...

    // Parentheses not needed in definition of macro itself
    #define prod(a, b, c)  \
        /* Parentheses needed in usage of macro arguments in macro body */ \
        (a) * (b) * (c)

    int
    fn(void)
    {
        /* Parentheses not needed in macro call strictly because it's a macro,
           though they might be needed for other reasons
        */
        return sum(1, 2 + 3, (4 + 5) * 6);
        /* As an example, without the parentheses in the macro body the above call would evaluate to
           1 * 2 + 3 * (4 + 5) * 6  which would give an unexpected result because it's implicitly
            (1 * 2) + (3 * (4 + 5) * 6) */
    }

Reply:
I mean like in the macro definition that you created here, where you're passing arguments directly and discretely from the macro to a called function. Parentheses are often needed in expressions in macros (both in definitions and in calls) than outside of macros, to ensure precedence and such. But I hope that
```
    controld_add_resource_history_xml_as(__func__, (parent), (rsc),     \
                                         (op), (node_name))
```

and

```
    controld_add_resource_history_xml_as(__func__, parent, rsc,     \
                                         op, node_name)
```

are exactly equivalent. ("Hope" since I've haven't used parentheses in those cases.)

Reply:
I believe you're correct, but C has enough obscure syntax that I'm not 100% sure :)